apps/pwa/vite.config.ts

```diff
@@ -1,10 +1,119 @@
 import path from "path";
-import react from "@vitejs/plugin-react";
 import { defineConfig } from "vite";
 import { createSvgIconsPlugin } from "vite-plugin-svg-icons";
+import reactSWC from "@vitejs/plugin-react-swc";
 import tsconfigPaths from "vite-tsconfig-paths";
+import { RegisterSWData, VitePWA, VitePWAOptions } from "vite-plugin-pwa";
+
+const manifestForPlugIn:Partial<VitePWAOptions> = {
+  devOptions: {
+    enabled: true,
+    type: "module",
+    navigateFallback: "index.html",
+  },
+  registerType: 'prompt',
+  workbox: {
+    globPatterns: ["**/*.{js,jsx,css,html,ico,png,svg,ts,tsx}"],
+    runtimeCaching: [
+      {
+        urlPattern: ({ request }: { request: Request }) => request.destination === "image",
+        handler: "StaleWhileRevalidate",
+      },
+      {
+        urlPattern: /\.(?:woff|woff2)$/,
+        handler: 'CacheFirst',
+        options: {
+          cacheName: 'fonts',
+          expiration: {
+            maxEntries: 30,
+            maxAgeSeconds: 60 * 60 * 24 * 365 // 1 year
+          },
+          cacheableResponse: {
+            statuses: [0, 200]
+          }
+        }
+      },
+      {
+        urlPattern: /\.(?:png|gif|jpg|jpeg|svg)$/,
+        handler: 'CacheFirst',
+        options: {
+          cacheName: 'images',
+          expiration: {
+            maxEntries: 60,
+            maxAgeSeconds: 30 * 24 * 60 * 60 // 30 days
+          }
+        }
+      }
+    ],
+  },
+  manifest: {
+    short_name: "Zeon",
+    name: "Zeon",
+    description: "Zeon is a platform for content creators to manage their content and monetize it.",
+    icons: [
+      {
+        src: "favicon.ico",
+        sizes: "48x48",
+        type: "image/x-icon",
+      },
+      {
+        src: "/favicon/favicon-16x16.png",
+        sizes: "16x16",
+        type: "image/png",
+      },
+      {
+        src: "/favicon/favicon-32x32.png",
+        sizes: "32x32",
+        type: "image/png",
```

Comment:
ye zeon nahi hai boierlpalte hai to, and because ye open source hai to koi bhi use kar sakega, to keep names as sample, don't use exact names

Reply:
okay, setting boilerplate data